contrib/epee/src/mlog.cpp

```diff
@@ -204,7 +204,8 @@ void mlog_configure(const std::string &filename_base, bool console, const std::s
           static_assert(std::is_integral<time_t>(), "bad time_t");
           return ta < tb;
         });
-        for (size_t i = 0; i < found_files.size() - max_log_files; ++i)
+        const int max_index_found_files = static_cast<int>(found_files.size()) - static_cast<int>(max_log_files); // Prevents unsigned var overflowing
+        for (int i = 0; i < max_index_found_files; ++i)
```

Comment:
What do you think of this change? This will prevent the index to overflow if the `max_log_files` is greater than `found_files.size()`
I also need an advice. Would simply writing:

`static_cast<int>(found_files.size() - max_log_files);`

be enough?

Reply:
I think you're missing the context here, all of this code is inside `if (found_files.size() >= max_log_files)`.

Reply:
> all of this code is inside if (found_files.size() >= max_log_files).

Ah, yeah. The `max_log_files > found_files.size()` case is already handled.

Reply:
What happens if the `found_files` is empty and `max_log_files` is zero?

Then the only line, preventing the code from crashing is [at line 174](https://github.com/monero-project/monero/pull/7480/files#diff-6676c0bac6e2626df8a81502a138684333631486fd1493787b383d37750e9890R174), located whole 30 lines above. So far away, that it seems out of context.

Reply:
We should settle at not touching this place at all, given the previous check `if (found_files.size() >= max_log_files)`. Don't just do changes because cppcheck said so, it's a dumb static analyzer. At most, you can replace `i <= found_files.size() - max_log_files` with `i  + max_log_files <= found_files.size()` if you worry about it.